### ISSUE TITLE
add timesplit classes

### ISSUE THREAD

**Issue by JochenSiegWork** [basf/MolPipeline#235]
Add classes for doing time splits.

Sklearn already has [TimeSeriesSplit](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html). However, this doesn't really cover our use cases because it splits in the direction of the time axis but based on number of samples.

What we need:

- Train/Test split based on specific date, like everything after 24.12.2025
- Train/Test split based on project start.
- Rolling time split: Like Sklearn's TimeSeries but splits's are determined by time, e.g. each fold is one year.
